src/linalg/rankUpdate.jl

```diff
@@ -61,6 +61,27 @@ function rankUpdate!(C::HermOrSym{T,Diagonal{T,Vector{T}}}, A::StridedMatrix{T},
     C
 end
 
+function rankUpdate!(C::HermOrSym{T,UniformBlockDiagonal{T}}, A::StridedMatrix{T}, α, β) where {T,S}
+    Cdat = C.data.data
+    @. Cdat = β * Cdat
+    nblks = size(Cdat, 3)
+    blksize = size(Cdat, 1)
+
+    for k in 1:nblks
+        ioffset = (k - 1) * blksize
+        joffset = (k - 1) * blksize
+        for i in 1:blksize, j in 1:blksize
+            iind = ioffset + i
+            jind = joffset + j
+            Arow = view(A, iind, :)
```

Comment:
I would probably write this as an explicit loop over `axes(A, 2)` and avoid the construction of views.

Reply:
So something like this:
```julia
AtAij = 0
for idx in axes(A, 2)
    # because the second is actually A', we swap index orders
    AtAij += A[iind,idx] * A[jind,idx]
end
Cdat[i,j,k] = Cdat[i,j,k] + α * AtAij
```
? 
I also thought about writing this as
```julia
AtAij = sum(A[iind,idx] * A[jind,idx] for idx in axes(A,2))
```
but then I realized the construction of the generator is probably no better (and maybe worse) than the construction of views.

Reply:
Actually what I had in mind is something like
```
for idx in axes(A, 2)
    Cdat[i,j,k] += α * A[iind, idx] * A[jind, idx]
end
```
I'm pretty sure the compiler will detect that the location `Cdat[i,j,k]` is a loop invariant.

Reply:
Ah, interesting. 

That might be a decent research project if we ever get an intern -- taking a look at `@code_llvm` and `@code_native` for some of our loops and seeing where the Julia compiler and LLVM are doing nice vectorization and other loop transformations for.

Reply:
An alternative is to resize `Cdat` to be a matrix with `blksize` rows.  Then the columns match up with the columns of `C`.  Also there is an intriguing function called `fldmod1` which makes it easy to step over the blocks but I have forgotten the details.  I'll take a look.

Reply:
If we're going to resize in that way -- would it make sense to change the internal storage used by `UniformBlockDiagonal` to always store in that format?

Reply:
There are times that you want the diagonal blocks to be accessed as k by k matrices.  The most important part is in evaluating `Lambda'Z'ZLambda + I`.  Also, the solving of multiple triangular systems to produce the `[i,1]` blocks of L for `i > 1`.  I think it is easier to resize from 3 dimensions to 2 when we need it, rather than the other way around.